Vagrantfile

```diff
@@ -1,42 +1,80 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
-HOSTNAME = 'ansible-st2'
-
 VAGRANTFILE_API_VERSION = '2'
+VIRTUAL_MACHINES = {
+  :u14 => {
+    :hostname => 'ubuntu14',
+  },
+  :u16 => {
+    :hostname => 'ubuntu16',
+  },
+  :centos6 => {
+    :hostname => 'centos6',
+  },
+  :centos7 => {
+    :hostname => 'centos7',
+  },
+}
+
+
 Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
-  config.vm.box = 'ubuntu/xenial64'
-  config.vm.hostname = "#{HOSTNAME}"
   config.vm.network "forwarded_port", guest: 22, host: 2200, auto_correct: true
   config.ssh.shell = "bash -c 'BASH_ENV=/etc/profile exec bash'"
+  config.ssh.forward_agent = true
 
-  config.vm.provider :virtualbox do |vb|
-    vb.name = "#{HOSTNAME}"
-    vb.customize ['modifyvm', :id, '--memory', '2048']
-  end
+  VIRTUAL_MACHINES.each do |name, cfg|
+    config.vm.define name do |vm_config|
+      vm_config.vm.hostname = cfg[:hostname]
 
-  if Vagrant.has_plugin?('vagrant-cachier')
-    config.cache.scope = :box
-  end
+      if name == :u14
+        vm_config.vm.box  = 'ubuntu/trusty64'
+      end
```

Comment:
We can include `box` types in `VIRTUAL_MACHINES` constant, what will allow us to simplify this:  
```rb
vm_config.vm.box = cfg[:box]
```